packages/test/test-version-utils/src/versionUtils.ts

```diff
@@ -281,15 +281,23 @@ export function internalSchema(publicVersion: string, internalVersion: string, r
     if (publicVersion === "2.0.0" && internalVersion < "2.0.0" && requested === -2) { return `^0.59.0-0`; }
     if (publicVersion === "2.0.0" && internalVersion === "2.0.0" && requested === -2) { return `^1.0.0-0`; }
 
+    // if the version number is for the older version scheme before 1.0.0
     if (publicVersion === "2.0.0" && internalVersion <= "2.0.0" && requested < -2) {
         const lastPrereleaseVersion = new semver.SemVer("0.59.0");
         const requestedMinorVersion = lastPrereleaseVersion.minor + (requested as number) + 2;
         return `^0.${requestedMinorVersion}.0-0`;
     }
 
     let parsedVersion;
+    let semverInternal: string = internalVersion;
+
+    if (internalVersion > publicVersion && requested <= -2) {
+        const version = internalVersion.split(".");
+        semverInternal = (parseInt(version[0], 10) + ((requested as number) + 1)).toString().concat(".0.0");
+    }
+
     try {
-        parsedVersion = new semver.SemVer(internalVersion);
+        parsedVersion = new semver.SemVer(semverInternal);
```

Comment:
I believe the issue is coming from the `return` on line 306. This value is being used to construct a Semver object, which then throws the error I've been seeing.